sdk/go/x/auto/local_workspace.go

```diff
@@ -46,6 +47,55 @@ type LocalWorkspace struct {
 	program         pulumi.RunFunc
 	envvars         map[string]string
 	secretsProvider string
+	pulumiVersion   Version
+}
+
+type Version struct {
+	Major  int
+	Minor  int
+	Patch  int
+	Suffix string
+}
+
+func (v Version) String() string {
+	if v.Suffix != "" {
+		return fmt.Sprintf("v%v.%v.%v-%s", v.Major, v.Minor, v.Patch, v.Suffix)
+	}
+	return fmt.Sprintf("v%v.%v.%v", v.Major, v.Minor, v.Patch)
+}
+
+func parseVersion(versionString string) (Version, error) {
```

Comment:
Any reason not to use https://github.com/blang/semver and one of it's `Parse` functions (`ParseTolerant`?) like we do elsewhere?

Also, wonder if we should just return `semver.Version` rather than defining our own `Version` struct. We already return it publicly elsewhere in the Go SDK, e.g.:

https://github.com/pulumi/pulumi/blob/b82b850af0d62461d47ce8948f62ef011c8fdd7b/sdk/go/pulumi/rpc.go#L703-L705

Reply:
Only so I would have the opportunity to do it all over again 😄